src/LEGO.AsyncAPI.Bindings/Sns/SnsOperationBinding.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using LEGO.AsyncAPI.Models.Any;
+using LEGO.AsyncAPI.Readers.ParseNodes;
+using LEGO.AsyncAPI.Writers;
+
+namespace LEGO.AsyncAPI.Bindings.Sns;
+
+public class SnsOperationBinding : OperationBinding<SnsOperationBinding>
+{
+    /// <summary>
+    /// Often we can assume that the SNS Topic is the channel name-we provide this field in case the you need to supply the ARN, or the Topic name is not the channel name in the AsyncAPI document.
+    /// </summary>
+    public Identifier Topic { get; set; }
+    
+    /// <summary>
+    /// The protocols that listen to this topic and their endpoints.
+    /// </summary>
+    public List<Consumer> Consumers { get; set; }
+    
+    /// <summary>
+    /// Policy for retries to HTTP. The field is the default for HTTP receivers of the SNS Topic which may be overridden by a specific consumer.
+    /// </summary>
+    public DeliveryPolicy DeliveryPolicy { get; set; }
+    
+    public override string BindingKey => "sns";
+
+    protected override FixedFieldMap<SnsOperationBinding> FixedFieldMap => new()
+    {
+        { "topic", (a, n) => { a.Topic = n.ParseMap(this.identifierFixFields); } },
+        { "consumers", (a, n) => { a.Consumers = n.CreateList(s => s.ParseMap(this.consumerFixedFields)); } },
+        { "deliveryPolicy", (a, n) => { a.DeliveryPolicy = n.ParseMap(this.deliveryPolicyFixedFields); } },
+    };
+
+    private FixedFieldMap<Identifier> identifierFixFields => new()
+    {
+        { "url", (a, n) => { a.Url = n.GetScalarValue(); } },
+        { "email", (a, n) => { a.Url = n.GetScalarValue(); } },
+        { "phone", (a, n) => { a.Url = n.GetScalarValue(); } },
+        { "arn", (a, n) => { a.Url = n.GetScalarValue(); } },
+        { "name", (a, n) => { a.Url = n.GetScalarValue(); } },
+    };
+
+    private FixedFieldMap<Consumer> consumerFixedFields => new ()
+    {
+        { "protocol", (a, n) => { a.Protocol = n.GetScalarValue().GetEnumFromDisplayName<Protocol>(); } },
+        { "endpoint", (a, n) => { a.Endpoint = n.ParseMap(this.identifierFixFields); } },
+        { "filterPolicy", (a, n) => { a.FilterPolicy = n.ParseMap(this.filterPolicyFixedFields); } },
+        { "rawMessageDelivery", (a, n) => { a.RawMessageDelivery = n.GetBooleanValue(); } },
+        { "redrivePolicy", (a, n) => { a.RedrivePolicy = n.ParseMap(this.redrivePolicyFixedFields); } },
+        { "deliveryPolicy", (a, n) => { a.DeliveryPolicy = n.ParseMap(this.deliveryPolicyFixedFields); } },
+        { "displayName", (a, n) => { a.DisplayName = n.GetScalarValue(); } },
+    };
+
+    private FixedFieldMap<FilterPolicy> filterPolicyFixedFields => new()
+    {
+        { "attributes", (a, n) => { a.Attributes = new AsyncApiObject(); }}
```

Comment:
@VisualBean How do I read in an arbitrary object like this? It seems to work perfectly for serialisation but I couldn't see anything in the library functions that looked promising for parsing it.

Reply:
Arbitrary object of what kind? JsonSchema or?

Reply:
There is a parsenode extension for reading the AnyType. It might still be internal though. Consider adding a validation rule (more about that in the wiki) to enforce the typings though. As IAny can also be integers which the attributes doesn't seem to support (from glancing over the binding pr)

Reply:
You should already be able to use 'Node.CreateAny()' for reading
<img width="272" alt="image" src="https://github.com/LEGO/AsyncAPI.NET/assets/5294032/5d8a9ad9-0776-4ca6-b8ba-9494180e6d05">

Reply:
Ive added an example of how to use 'Any' in the `CustomBinding_Should.cs` file

Reply:
This worked! 😁